### PR TITLE
Parse favorite attribute for OCShare on NC 25+

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
TODO
- [x] wait for https://github.com/nextcloud/server/pull/34133 to be merged
- [x] wait for https://github.com/nextcloud/server/pull/34223

Requires: https://github.com/nextcloud/server/pull/34133

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

File: library/src/main/java/com/owncloud/android/lib/resources/shares/GetShareRemoteOperation.java [hunk 1]
```diff
@@ -61,7 +61,9 @@ protected RemoteOperationResult<List<OCShare>> run(OwnCloudClient client) {
 
         // Get the response
         try {
-            get = new GetMethod(client.getBaseUri() + ShareUtils.SHARING_API_PATH + "/" + remoteId);
+            get = new GetMethod(client.getBaseUri() +
+                    ShareUtils.SHARING_API_PATH
+                    + "/" + remoteId + "?includeTags=true");
```
Comment (AlvaroBrey) [inline library/src/main/java/com/owncloud/android/lib/resources/shares/GetShareRemoteOperation.java:66]: `includeTags=true` should be in a constant somewhere and shared between this class and `GetSharesRemoteOperation`

File: library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java [hunk 2]
```diff
@@ -163,7 +166,9 @@ public ArrayList<OCShare> parseXMLResponse(InputStream is) throws XmlPullParserE
 			parser.setInput(is, null);
 			parser.nextTag();
 			return readOCS(parser);
-
+		} catch (Exception e) {
+			Log_OC.e(this, "Error parsing xml", e);
+			return new ArrayList<>();
```
Comment (AlvaroBrey) [inline library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java:171]: Judging from the method signature, `parseXMLResponse` is supposed to throw exceptions, not catch them and return an empty list.

File: library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java [hunk 3]
```diff
@@ -418,6 +423,17 @@ private void readElement(XmlPullParser parser, ArrayList<OCShare> shares)
 					share.setOwnerDisplayName(readNode(parser, NODE_DISPLAYNAME_FILE_OWNER));
 					break;
 
+				case NODE_TAGS:
+					List<String> tags = readArrayNode(parser);
+					for (String tag : tags) {
+						if (tag.startsWith("_$!<Favorite>")) {
```
Comment (AlvaroBrey) [inline library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java:429]: The `startsWith` content should be a constant

**Review by AlvaroBrey — CHANGES_REQUESTED** [review]
Some minor changes requested, also please take a look at Detekt and Spotbugs

File: library/src/main/java/com/owncloud/android/lib/resources/shares/GetShareRemoteOperation.java (hunk 1 above)
Comment (tobiasKaminsky) [reply library/src/main/java/com/owncloud/android/lib/resources/shares/GetShareRemoteOperation.java:66]: Was quickly hardcoded for testing and then forgotten…Thanks!

File: library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java (hunk 2 above)
Comment (tobiasKaminsky) [reply library/src/main/java/com/owncloud/android/lib/resources/shares/ShareXMLParser.java:171]: Totally true!

**Review by AlvaroBrey — DISMISSED** [review]

**Comment by AlvaroBrey** [comment]
@tobiasKaminsky as this is kinda old I didn't wanna merge without you having a look at it, feel free to merge when  you want

**Comment by tobiasKaminsky** [comment]
I created a branch on files for it: https://github.com/nextcloud/android/pull/11138

**Review by AlvaroBrey — DISMISSED** [review]

**Review by AlvaroBrey — APPROVED** [review]

**Pull request merged**